### PR TITLE
enforce package deprecation checks

### PR DESCRIPTION
**Pull request by mario-guerra**

### CODE REVIEW

File: eng/scripts/Update-Release-Versions.ps1 [hunk 1]
```diff
@@ -309,6 +309,41 @@ function CheckAll($langs)
...
+        if (!$pkg.ReplaceGuide)
+        {
+          Write-Warning "No migration guide set for deprecated package '$($pkg.Package)' in $csvFile."
+          Write-Warning "Migration guide link must adhere to the following convention 'aka.ms/azsdk/<language>/migrate/<library>'"
+          $foundIssues = $true
+          if ($pkg.EOLDate -gt $migrationGuideStandardEnforcementDate)
+          {
+            if ($pkg.ReplaceGuide -notmatch $migrationGuideLinkFormat)
+            {
+              Write-Warning "Migration guide link does not adhere to convention 'aka.ms/azsdk/<language>/migrate/<library>'"
```
Comment (weshaggard) [inline eng/scripts/Update-Release-Versions.ps1:340]: Do we want to enforce this format?

File: eng/scripts/Update-Release-Versions.ps1 [hunk 2]
```diff
@@ -309,6 +309,41 @@ function CheckAll($langs)
   {
     $clientPackages, $_ = Get-PackageListForLanguageSplit $lang
```
Comment (weshaggard) [inline eng/scripts/Update-Release-Versions.ps1:310]: Can you please undo the changes to the csv files? We should keep those empty quotes.

Comment (weshaggard) [reply eng/scripts/Update-Release-Versions.ps1:310]: Also remove the output.txt file

**Comment by mario-guerra** [comment]
Closing this PR. I will resubmit a new PR after fixing the issues I introduced in the CSV files.
